File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/go_solutions/3243-shortest-distance-after-road-addition-queries-i.go
func shortestDistance(n int, edges [][]int, queries [][]int) []int {
	dist := make([][]int, n)
	for i := 0; i < n; i++ {
		dist[i] = make([]int, n)
		for j := 0; j < n; j++ {
			if i == j {
				dist[i][j] = 0
			} else {
				dist[i][j] = 1000000000
			}
		}
	}

	for _, edge := range edges {
		u, v := edge[0], edge[1]
		dist[u][v] = 1
		dist[v][u] = 1
	}

	for k := 0; k < n; k++ {
		for i := 0; i < n; i++ {
			for j := 0; j < n; j++ {
				dist[i][j] = min(dist[i][j], dist[i][k]+dist[k][j])
			}
		}
	}

	result := make([]int, len(queries))
	for i, query := range queries {
		u, v := query[0], query[1]
		if dist[u][v] >= 1000000000 {
			result[i] = -1
		} else {
			result[i] = dist[u][v]
		}
	}

	return result
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

File: leetcode/AIcoding/2.Gemini-2.0/python3_solutions/3243-shortest-distance-after-road-addition-queries-i.py
class Solution:
    def shortestDistance(self, n: int, edges: list[list[int]], queries: list[list[int]]) -> list[int]:
        dist = [[float('inf')] * n for _ in range(n)]
        for i in range(n):
            dist[i][i] = 0
        for u, v in edges:
            dist[u][v] = 1
            dist[v][u] = 1
        for k in range(n):
            for i in range(n):
                for j in range(n):
                    dist[i][j] = min(dist[i][j], dist[i][k] + dist[k][j])
        ans = []
        for u, v, w in queries:
            temp_dist = [row[:] for row in dist]
            for i in range(n):
                for j in range(n):
                    temp_dist[i][j] = min(temp_dist[i][j], temp_dist[i][u] + w + temp_dist[v][j], temp_dist[i][v] + w + temp_dist[u][j])
            min_dist = float('inf')
            for i in range(n):
                for j in range(i + 1, n):
                    min_dist = min(min_dist, temp_dist[i][j])
            if min_dist == float('inf'):
                ans.append(-1)
            else:
                ans.append(min_dist)
        return ans